website/project/model.py

```diff
@@ -309,6 +334,17 @@ def edit(self, content, auth, save=False):
         self.content = content
         self.modified = True
         self.date_modified = datetime.datetime.utcnow()
+
+        if (new_mentions):
+            self.new_mentions = new_mentions
+            old_mentions = self.old_mentions
+            validate = lambda m: m not in old_mentions and validate_contributor(m, self.node.contributors)
+            self.new_mentions = filter(validate, self.new_mentions)
+            if len(self.new_mentions) > 0:
+                if save:
+                    project_signals.mention_added.send(self, auth=auth)
+                    self.old_mentions.extend(self.new_mentions)
```

Comment:
Let's move this whole `if new_mentions` block down under the `if save` condition just below, rather than structuring with the additional `if save` here.